controller/controller.go

```diff
@@ -937,7 +938,7 @@ func saveInstallResults(rootDir string, md *model.SystemInstall) error {
 		log.Error("Failed to log Telemetry success record")
 	}
 
-	if md.PostArchive {
+	if md.PostArchive.Value() {
```

Comment:
Question:
Are you fixing the PostArchive as well to use a boolset. I am guessing this one does not really have to do with Auto-update bug itself. Correct?

Reply:
Correct, it is just the other "defaults to true" model boolean that also benefits from the new data type.

Reply:
I would rather have it moved to a new commit to keep it clean

Reply:
pushed